app/pages/SummaryPage.scala

```diff
@@ -79,102 +100,134 @@ case object SummaryPage {
+      if (userAnswers.flatMap(_.get(StructureBuildingQualifyingDatePage(0))).isDefined) {
+        TaskListTag.InProgress
+      } else {
+        TaskListTag.NotStarted
+      },
       "structuresAndBuildingAllowance_link"
     )
-  }
 
-  private def propertyAllowancesItem(taxYear: Int) = {
-    TaskListItem("summary.allowances",
+  private def propertyAllowancesItem(taxYear: Int) =
+    TaskListItem(
+      "summary.allowances",
       controllers.allowances.routes.AllowancesStartController.onPageLoad(taxYear),
       TaskListTag.NotStarted,
       "allowances_link"
     )
-  }
 
-  private def propertyRentalsExpensesItem(userAnswers: Option[UserAnswers], taxYear: Int) = {
-    TaskListItem("summary.expenses",
+  private def propertyRentalsExpensesItem(userAnswers: Option[UserAnswers], taxYear: Int) =
+    TaskListItem(
+      "summary.expenses",
       controllers.propertyrentals.expenses.routes.ExpensesStartController.onPageLoad(taxYear),
-      if (userAnswers.flatMap(_.get(ConsolidatedExpensesPage)).isDefined) TaskListTag.InProgress else TaskListTag.NotStarted,
+      if (userAnswers.flatMap(_.get(ConsolidatedExpensesPage)).isDefined) {
+        TaskListTag.InProgress
+      } else {
+        TaskListTag.NotStarted
+      },
       "expenses_link"
     )
-  }
 
-  private def propertyRentalsIncomeItem(userAnswers: Option[UserAnswers], taxYear: Int) = {
+  private def propertyRentalsIncomeItem(userAnswers: Option[UserAnswers], taxYear: Int) =
     TaskListItem(
       "summary.income",
       controllers.propertyrentals.income.routes.PropertyIncomeStartController.onPageLoad(taxYear),
-      if (userAnswers.flatMap(_.get(ClaimPropertyIncomeAllowancePage)).isDefined) TaskListTag.InProgress else TaskListTag.NotStarted,
+      if (userAnswers.flatMap(_.get(ClaimPropertyIncomeAllowancePage)).isDefined) {
+        TaskListTag.InProgress
+      } else {
+        TaskListTag.NotStarted
+      },
       "income_link"
     )
-  }
 
-  private def propertyAboutItem(userAnswers: Option[UserAnswers], taxYear: Int) = {
+  private def propertyAboutItem(userAnswers: Option[UserAnswers], taxYear: Int) =
     TaskListItem(
       "summary.about",
       controllers.propertyrentals.routes.PropertyRentalsStartController.onPageLoad(taxYear),
       if (userAnswers.flatMap(_.get(TotalIncomePage)).isDefined) TaskListTag.InProgress else TaskListTag.NotStarted,
       "about_link"
     )
-  }
 
-  private def ukRentARoomAboutItem(userAnswers: Option[UserAnswers], taxYear: Int) = {
+  private def ukRentARoomAboutItem(userAnswers: Option[UserAnswers], taxYear: Int) =
     TaskListItem(
       "summary.about",
-      controllers.ukrentaroom.routes.UkRentARoomJointlyLetController.onPageLoad(taxYear, NormalMode),
-      if (userAnswers.flatMap(_.get(UkRentARoomJointlyLetPage)).isDefined) TaskListTag.InProgress else TaskListTag.NotStarted,
+      controllers.ukrentaroom.routes.UkRentARoomJointlyLetController.onPageLoad(taxYear, NormalMode), {
```

Comment:
@modivilioglu I have changed this a tad bit to account for the what the user has selected in the last page. The logic that you have implemented wrt whether the UkRentARoomJointlyLetPage exists in the user answer exists is still there but is overridden if the user has selected that they have completed this section and task is tagged as Completed. Please have a look a let me know what you think